lib/MemoryAnalysisModules/SCEVAA.cpp

```diff
@@ -42,6 +43,8 @@ class SCEVAA : public ModulePass, public liberty::ClassicLoopAA {
   static bool alwaysGreaterThan(ScalarEvolution *SE, const SCEV *difference,
                                 const Loop *L, const APInt &positive,
                                 const APInt &negative) {
+    if(difference->getSCEVType() == llvm::scCouldNotCompute)
```

Comment:
This looks like something that should be promoted to the caller? And when would a SCEV not able to be computed? Is that a bug? If so, should be an assert rather than conservative handling.